### DISCUSSION TITLE
How do you define a custom node in svelte 5?

### DISCUSSION THREAD

**Discussion by surrealroad** [xyflow/xyflow#4832] [Q&A]
I've been chasing my tail a lot over this. I cannot for the life of me figure out how to define a custom node (using typescript) with svelte 5. Can anyone point me to an example?

**Comment by bcakmakoglu** [comment]
Doesn't using the `nodesTypes` prop work for you? I don't see how Svelte 5 makes any difference in this 🤔

[There's a custom node example here.](https://svelteflow.dev/examples/nodes/custom-node)

**Comment by surrealroad** [comment]
Well I couldn't figure out how to update the following
```
type $$Props = NodeProps;
 
export let data: $$Props['data'];
export let isConnectable: $$Props['isConnectable'];
$$restProps
```

But after playing around for a bit, I think doing something like 
```
type EditableTextNode = Node<EditableTextNodeData>;
interface EditableTextNodeData extends Record<string, unknown> {
		onChange?: (data: { content: string }) => void;
}
interface Props extends NodeProps<EditableTextNode> {
		data: EditableTextNodeData;
}

let { data = $bindable(), isConnectable, selected, ...rest }: Props = $props();
```

Should work
